### PR TITLE
Add SystemCredential and Environment

### PR DESCRIPTION
**Pull request by timotheev**
More authentication options for the API functions.

### CODE REVIEW

File: Source/APIFunctions/APIs/Common.wl [hunk 1]
```diff
@@ -141,6 +146,18 @@ ConnectToService[name_, authentication_] :=
 		APIFailure
 	]];
 
+ConnectToService[name_] := GU`Scope @ Enclose[
+	DBPrint["ConnectToService: ", StringForm["Looking for any available connexion to ``.", name]];
+	so = ConnectToService[name, "Available"];
```
[the remark this answers is not in the document]
Comment (timotheev) [reply Source/APIFunctions/APIs/Common.wl]: Done

File: Source/APIFunctions/APIs/Common.wl [hunk 2]
```diff
@@ -122,13 +127,13 @@ ConnectToService[name_, authentication_] :=
...
-				Confirm @ TestConnection[name, temp];
+				DBPrint["ConnectToService: ", StringForm["Using provided connexion w/o validation ``.", Last @ authentication]];
 				key = Confirm @ getAuthenticationHash[temp];
 				,
-			"Wolfram" | _,
-				(**)
+			_,
+				(* TODO : "WolframCloud " *)
 				DBPrint["ConnectToService: ", StringForm["Unsupported authentication ``.", authentication]];
 				GU`ThrowFailure["bdauth", "Wolfram"]
```
[the remark this answers is not in the document]
Comment (timotheev) [reply Source/APIFunctions/APIs/Common.wl]: Done.

File: Source/APIFunctions/APIs/Common.wl [hunk 3]
```diff
@@ -122,13 +127,13 @@ ConnectToService[name_, authentication_] :=
...
 				temp = authentication;
-				DBPrint["ConnectToService: ", StringForm["Validating connexion ``.", Last @ authentication]];
-				Confirm @ TestConnection[name, temp];
+				DBPrint["ConnectToService: ", StringForm["Using provided connexion w/o validation ``.", Last @ authentication]];
 				key = Confirm @ getAuthenticationHash[temp];
 				,
-			"Wolfram" | _,
-				(**)
+			_,
+				(* TODO : "WolframCloud " *)
```
[the remark this answers is not in the document]
Comment (timotheev) [reply Source/APIFunctions/APIs/Common.wl]: Done.

File: Source/APIFunctions/APIs/Common.wl [hunk 4]
```diff
@@ -82,55 +83,56 @@ SetAttributes[errorSymbol, HoldFirst]
...
+	"Dialog";
 
 ClearAll["ConnectToService"]
 ConnectToService[name_, authentication_] :=
 	GU`Scope[Enclose[
 		Switch[authentication,
-			"Available",
-				temp = SelectFirst[
+			"Cached",
+				so = Confirm @ SelectFirst[
```
Comment (batracos) [inline Source/APIFunctions/APIs/Common.wl:94]: This should be `temp` to be retrieved at the end of the function.

**Pull request merged**